### PR TITLE
Switch to PyPI Trusted Publishing, and remove twine

### PR DESCRIPTION
**Pull request by GenevieveBuckley**
Closes https://github.com/napari/napari-animation/issues/211

This PR switches to PyPI deployment with Trusted Publishing (see [the announcement](https://blog.pypi.org/posts/2023-04-20-introducing-trusted-publishers/) and [how trusted publishing works](https://docs.pypi.org/trusted-publishers/internals/)). It removes twine, and also removes the need for PyPI API tokens being used as github secrets.

This PR is not sufficient on its own, someone else also needs to:
* [Add a trusted publisher to our existing PyPI project](https://docs.pypi.org/trusted-publishers/adding-a-publisher/) (or you can [create a new PyPI project with a trusted publisher](https://docs.pypi.org/trusted-publishers/creating-a-project-through-oidc/)).
* Delete the old PyPI API tokens, from both PyPI and the GitHub repository secrets settings.

### CODE REVIEW

File: .github/workflows/test_and_deploy.yml [hunk 1]
```diff
@@ -84,10 +84,16 @@ jobs:
...
+    # and requires that you have setup PyPI Trusted Publishing
+    # (see https://docs.pypi.org/trusted-publishers/adding-a-publisher/)
     needs: [ test ]
     runs-on: ubuntu-latest
+    permissions:
+      # IMPORTANT: this permission is mandatory for trusted publishing
+      id-token: write
+    environment:
+      name: pypi
+      url: https://pypi.org/p/<your-pypi-project-name>
```
Comment (GenevieveBuckley) [inline .github/workflows/test_and_deploy.yml:96]:
```suggestion
      url: https://pypi.org/project/napari-animation
```

**Comment by GenevieveBuckley** [comment]
PR marked as draft, so it doesn't get merged before someone [enables trusted publishing on PyPI](https://docs.pypi.org/trusted-publishers/adding-a-publisher/)

File: .github/workflows/test_and_deploy.yml [hunk 2]
```diff
@@ -98,12 +104,11 @@ jobs:
...
-          pip install -U setuptools setuptools_scm wheel twine
-      - name: Build and publish
-        env:
-          TWINE_USERNAME: __token__
-          TWINE_PASSWORD: ${{ secrets.TWINE_API_KEY }}
+          pip install -U setuptools setuptools_scm wheel
+      - name: Build python package
         run: |
           git tag
           python setup.py sdist bdist_wheel
```
Comment (GenevieveBuckley) [inline .github/workflows/test_and_deploy.yml:111]: Also, do you want to switch to using `python -m build` too?

Comment (Czaki) [reply .github/workflows/test_and_deploy.yml:111]: Yes, please do this

**Comment by alisterburt** [comment]
Just run it and see, no big deal if it fails!

**Comment by jni** [comment]
it's enabled.

**Comment by jni** [comment]
> Just run it and see, no big deal if it fails!

[loss aversion](https://www.discovermagazine.com/mind/why-we-hate-the-feeling-of-failure-and-how-it-affects-us) is powerful stuff. 😂 

https://www.nytimes.com/interactive/2015/07/03/upshot/a-quick-puzzle-to-test-your-problem-solving.html

File: .github/workflows/test_and_deploy.yml (hunk 2 above)
Comment (GenevieveBuckley) [inline .github/workflows/test_and_deploy.yml:111]:
```suggestion
          python -m build
```

File: .github/workflows/test_and_deploy.yml [hunk 3]
```diff
@@ -98,12 +104,11 @@ jobs:
       - name: Install dependencies
         run: |
           python -m pip install --upgrade pip
-          pip install -U setuptools setuptools_scm wheel twine
-      - name: Build and publish
-        env:
-          TWINE_USERNAME: __token__
-          TWINE_PASSWORD: ${{ secrets.TWINE_API_KEY }}
+          pip install -U setuptools setuptools_scm wheel
```
Comment (GenevieveBuckley) [inline .github/workflows/test_and_deploy.yml:107]:
```suggestion
          python -m pip install build setuptools setuptools_scm
```

**Comment by GenevieveBuckley** [comment]
CI tests failing, a fix is in progress here: https://github.com/napari/napari-animation/pull/214

**Comment by alisterburt** [comment]
> > Just run it and see, no big deal if it fails!
> 
> [loss aversion](https://www.discovermagazine.com/mind/why-we-hate-the-feeling-of-failure-and-how-it-affects-us) is powerful stuff. 😂
> 
> https://www.nytimes.com/interactive/2015/07/03/upshot/a-quick-puzzle-to-test-your-problem-solving.html

😆 thanks for this! Interesting read :)

File: pyproject.toml [hunk 4]
```diff
@@ -53,5 +53,5 @@ ignore = [
     "SIM117", # flake8-simplify - some of merged with statements are not looking great with black, reanble after drop python 3.9
 ]
 
-target-version = "py38"
+target-version = "py312"
```
Comment (GenevieveBuckley) [inline pyproject.toml:56]:
```suggestion
target-version = "py39"
```

**Comment by GenevieveBuckley** [comment]
PR https://github.com/napari/napari-animation/pull/218 should fix the ruff CI errors

**Comment by jni** [comment]
> 😆 thanks for this! Interesting read :)

@alisterburt did you pass the test? I am sad to admit that I was one of the 77% of chumps who confidently guessed powers of 2 after a bunch of positive tests. 😬 

The crazy thing is that I consciously think about that test/article often, but still suffer from it. I'll spend so much time setting things up just-so, even when there is literally no cost to failure. But I deeply don't want the computer-says-no message! Stupid brain! 🧠 😂

**Comment by jni** [comment]
@GenevieveBuckley wanna merge in main? 🙏 Then this should be ready to go in! 🚀

File: pyproject.toml [hunk 5]
```diff
@@ -53,5 +53,5 @@ ignore = [
     "SIM117", # flake8-simplify - some of merged with statements are not looking great with black, reanble after drop python 3.9
 ]
 
-target-version = "py38"
+target-version = "py39"
```
Comment (GenevieveBuckley) [inline pyproject.toml:56]:
```suggestion
target-version = "py38"
```

**Comment by GenevieveBuckley** [comment]
Ready to merge when the CI checks are all green.
Then we'll need someone to test a release candidate, to see if the PyPI trusted publishing works.

**Pull request merged**